posts/GSoC 2019.md

```diff
@@ -0,0 +1,64 @@
+## What Errors Can Be Visualized or Fixed:
+The errors being detected by the application will include but not limited:
+Syntax errors
+
+-   duplicate lines
+-   missing digits from end of line
+-   missing 'E'
+-   whatever else we find
+
+Chemistry errors
+
+-   discontinuous polynomicals
+-   exceeding collision limit
+-   inconsistent thermo for explicit reverse rates
+-   negative rate coefficient (as a  [result](https://github.com/comocheng/wiki/issues/375#)  of sum of arrhenius or PLOG expressions, eg.  [here](https://github.com/Cantera/cantera-website/issues/77))
+-   excess stiffness
+-   making  [error](https://github.com/comocheng/wiki/issues/375#)  messages from CVODES easier to understand
+-   making errors from CVODES easier to catch (raise a python Error)
+-   dead-end pathways
+## How Can I Achieve It
+
+The debugging tool PyCanChemAndYouCan2 is a web-based application developed in Django 2.2.  Users can upload original files and download revised files from the website, and the application contains  ACE code editor to allow users edit their files on the website. For Debugging script, Cantera has `ck2cti.py` file to convert different file format, so the debugging part will be test different defective mechanism files and update `ck2cti.py` file. The ideal condition is that the application can help users find the errors and sugget the fixes or fix errors automatically. However, it might be a long-term goal, so we make some minimum goal for this GSoC project.
+Here are the basic ideas about the project:
+ -   a script that takes in an input file and returns an input file with added comments, which describe detected errors.
+-   a web site that makes it easy to interact with that script (which runs on the web server), and renders the comments in an attractive way
+
+The idea is that the script can be used, developed, tested, as a stand-alone thing without the website.  
+The website just makes things look prettier. eg. a custom "syntax highlighting" mode that detects comments describing certain problems and renders them in red, or whatever.
+
+Here is the workflow of the application:
+
+```
+A[upload files] --> B[list files] --> C{choose one and execute by ck2cte.py}  --> D{show input file with added comments on ACE editor}
```

Comment:
Why do some of these have square brackets `[]` and some have curly brackets `{}`? What are the letters `A`, `B`, etc. for? You have a small typo, `ck2cte.py`